added multi zool level support for GC and its usage

amartya4256 opened the pull request.

simon-spinner committed:
Fix getWindowPointerPosition on Wayland #177

Depending on the window currently focused the
coordinates returned may be wrong (sometimes even
negative). This happens mainly with popup windows.

Fixes https://github.com/eclipse-platform/eclipse.platform.swt/issues/177

akoch-yatta committed:
Add opt-in method to propagate zoom changes

This contribution adds the base logics to support the dynamical adaption on SWT applications on DPI related zoom changes in the win32 implementation. It consists of:
- adds "swt.autoScale.updateOnRuntime"-flag to enable updating the widgets on runtime on DPI zoom changes
- Component to register callbacks on DPI changes. Each callback should be responsible for the attributes of its class

Contributes to https://github.com/eclipse-platform/eclipse.platform.swt/issues/62
and https://github.com/eclipse-platform/eclipse.platform.swt/issues/127

akoch-yatta committed:
Provide utilities to allow dpi dependent zooming of fonts

This contribution extends the support for dynamical adaption on SWT applications on DPI related zoom changes in the win32 implementation by adding support to provider and cache scaled variants of Font. It consists of:
- internal font registries to enable reusage of scaled variants of fonts. If a font is disposed from external, it will be automatically recreated on demand. There are two registries:
- a default registry to mimic the existing behavior, that all fonts are scaled to the native zoom of the primary monitor
- a registry to create and manage scaled variants of fonts
The SWTFontProvider is configured based on the "swt.autoScale.updateOnRuntime"-flag to use one of the two font registries. The creation of system fonts is adapted to use the provider

Contributes to https://github.com/eclipse-platform/eclipse.platform.swt/issues/62
and https://github.com/eclipse-platform/eclipse.platform.swt/issues/127

akoch-yatta committed:
Adapt calls for image bounds to consider the actual zoom of the widget

This contribution replaces calls to Image::getBoundsInPixels in the widgets in the win32 implementation with calls to consider the actual zoom of the affected widget.

Contributes to https://github.com/eclipse-platform/eclipse.platform.swt/issues/62
and https://github.com/eclipse-platform/eclipse.platform.swt/issues/127

akoch-yatta committed:
FIX: Rescaling a rectangle should only consider current and target zoom

As current and target zoom are given as parameters, the scaling must not be skipped if the cached value in deviceZoom equals 100, but only the given values must be used for it.

HannesWell committed:
[Build|Win32] Simplify build.bat for SWT's windows native binaries

Also remove elements specific to old build-clusters that are unused now.

deepika-u committed:
Revert "Revert "Set the selection to 1st CTabItem when it is created." (#1087)"

This reverts commit 2a51e8c925354594e42fc758b332336db937e01c.

fedejeanne committed:
Fix Maven launch config to build SWT binaries and change Readme

- Change base directory of launch config
- Activate the option "Refresh" in launch config

Co-authored-by: Hannes Wellmann <wellmann.hannes1@gmx.net>

amartya4256 committed:
Multi zoom level support of GC for win32

This commit contributes to multi zoom level support of GC which can be extended by other resources and widgets.

Contributes to
https://github.com/eclipse-platform/eclipse.platform.swt/issues/62 and https://github.com/eclipse-platform/eclipse.platform.swt/issues/131

fedejeanne committed:
Update documentation on how to build SWT binaries

- Move the part about building and testing to the general Readme file
since it can now be done in the same way in Windows, Mac and Linux.
- Adapt the part about configuring the workspace to run snippets in
Windows.
- Move the "Development overview" to the general Readme file
- Move the part about running the snippets to the Readme.md and change
it so it mentions the pre-configured IDE installation (it's way easier)
- Also add the perma-link to the section of the POM that calls Ant
- Mention GTK4 in Linux and link to official website
- Provide links to Adoptium's JDK 17 in all three readmes
- Remove the part about setting SWT_JAVA_HOME in all 3 platforms

Co-authored-by: Hannes Wellmann <wellmann.hannes1@gmx.net>

jukzi committed:
fix IllegalArgumentException when deleting text while IME open #1162

If text is deleted while the IME assistent is open the start of the open
IME has to be adjusted accordingly

https://github.com/eclipse-platform/eclipse.platform.swt/issues/1162

jukzi committed:
[win] GetDpiForMonitor: prevent devision by zero #193

https://github.com/eclipse-platform/.github/issues/193

dependabot[bot] committed:
Bump andymckay/append-gist-action from 0.2 to 0.3

Bumps [andymckay/append-gist-action](https://github.com/andymckay/append-gist-action) from 0.2 to 0.3.
- [Release notes](https://github.com/andymckay/append-gist-action/releases)
- [Commits](https://github.com/andymckay/append-gist-action/compare/6e8d64427fe47cbacf4ab6b890411f1d67c07f3e...ab30bf28df67017c7ad696500b218558c7c04db3)

---
updated-dependencies:
- dependency-name: andymckay/append-gist-action
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

akoch-yatta committed:
Apply DPI zoom changes to basic controls

This contribution adds and registers a DPI change handler for all widgets in the org.eclipse.swt.widgets package of the win32 implementation. These handler will be registered only, when the "swt.autoScale.updateOnRuntime"-flag is set to true. If a DPI change is detected they will be called accordingly.

Contributes to https://github.com/eclipse-platform/eclipse.platform.swt/issues/62
and https://github.com/eclipse-platform/eclipse.platform.swt/issues/127

dependabot[bot] committed:
Bump eclipsecbi/fedora-gtk3-mutter in /container

Bumps eclipsecbi/fedora-gtk3-mutter from `91dce77` to `54da20a`.

---
updated-dependencies:
- dependency-name: eclipsecbi/fedora-gtk3-mutter
  dependency-type: direct:production
...

Signed-off-by: dependabot[bot] <support@github.com>

wiresketch committed:
Fix Scrollable.getScrollbarsMode to be consistent #1182

Fix Scrollable.getScrollbarsMode to return SWT.NONE when
GTK_OVERLAY_SCROLLING=0 setting is present for Linux/GTK3.

Fixes: #1182

dependabot[bot] committed:
Bump actions/upload-artifact from 4.3.1 to 4.3.2

Bumps [actions/upload-artifact](https://github.com/actions/upload-artifact) from 4.3.1 to 4.3.2.
- [Release notes](https://github.com/actions/upload-artifact/releases)
- [Commits](https://github.com/actions/upload-artifact/compare/5d5d22a31266ced268874388b861e4b58bb5c2f3...1746f4ab65b179e0ea60a494b83293b640dd5bba)

---
updated-dependencies:
- dependency-name: actions/upload-artifact
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

eneufeld committed:
[win][edge] add focus delegate to browser

Currently when the ICoreWebView2Controller gets the focus this is not
reflected on the browser widget.
This adds a GotFocus handler and forces the focus on the browser widget.

wiresketch committed:
Add Scrollable.setScrollbarsMode for changing overlay mode

Add Scrollable.setScrollbarsMode(int) API to provide the possibility
to disable/enable overlay scrollbars.

Fixes https://github.com/eclipse-platform/eclipse.platform.swt/issues/1184

dependabot[bot] committed:
Bump eclipsecbi/fedora-gtk3-mutter in /container

Bumps eclipsecbi/fedora-gtk3-mutter from `92c8ca8` to `2a6899b`.

---
updated-dependencies:
- dependency-name: eclipsecbi/fedora-gtk3-mutter
  dependency-type: direct:production
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
Bump actions/checkout from 4.1.2 to 4.1.3

Bumps [actions/checkout](https://github.com/actions/checkout) from 4.1.2 to 4.1.3.
- [Release notes](https://github.com/actions/checkout/releases)
- [Changelog](https://github.com/actions/checkout/blob/main/CHANGELOG.md)
- [Commits](https://github.com/actions/checkout/compare/9bb56186c3b09b4f86b1c65136769dd318469633...1d96c772d19495a3b5c517cd2bc0cb401ea0529f)

---
updated-dependencies:
- dependency-name: actions/checkout
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
Bump actions/upload-artifact from 4.3.2 to 4.3.3

Bumps [actions/upload-artifact](https://github.com/actions/upload-artifact) from 4.3.2 to 4.3.3.
- [Release notes](https://github.com/actions/upload-artifact/releases)
- [Commits](https://github.com/actions/upload-artifact/compare/1746f4ab65b179e0ea60a494b83293b640dd5bba...65462800fd760344b1a7b4382951275a0abb4808)

---
updated-dependencies:
- dependency-name: actions/upload-artifact
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

akoch-yatta committed:
Extend DPIUtil to provide scale up and down methods with target zoom as parameter

This contribution adds additional methods to scale different datatypes up or down by passing the target zoom level as parameter. All existing methods will delegate to the added methods and pass DPIUtil::deviceZoom as zoom.

Contributes to https://github.com/eclipse-platform/eclipse.platform.swt/issues/62
and https://github.com/eclipse-platform/eclipse.platform.swt/issues/131

Commit:
v4965r7

Commit:
v4965r8

dependabot[bot] committed:
Bump actions/checkout from 4.1.3 to 4.1.4

Bumps [actions/checkout](https://github.com/actions/checkout) from 4.1.3 to 4.1.4.
- [Release notes](https://github.com/actions/checkout/releases)
- [Changelog](https://github.com/actions/checkout/blob/main/CHANGELOG.md)
- [Commits](https://github.com/actions/checkout/compare/1d96c772d19495a3b5c517cd2bc0cb401ea0529f...0ad4b8fadaa221de15dcec353f45205ec38ea70b)

---
updated-dependencies:
- dependency-name: actions/checkout
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
Bump eclipsecbi/fedora-gtk3-mutter in /container

Bumps eclipsecbi/fedora-gtk3-mutter from `2a6899b` to `317e341`.

---
updated-dependencies:
- dependency-name: eclipsecbi/fedora-gtk3-mutter
  dependency-type: direct:production
...

Signed-off-by: dependabot[bot] <support@github.com>

fedejeanne committed:
Apply clean-up: Remove unnecessary and unused code

- Remove unused imports
- Remove unnecessary casts
- Remove redundant semicolons
- Remove unnecessary '$NON-NLS$' tags

HeikoKlare committed:
Adapt GitHub Actions configuration to execute MacOS on x86_64 image

GitHub Actions changed their latest images for MacOS to use the aarch64
instead of the x86_64 architecture. The SWT tests are currently failing
because they still assume macos-latest to be an x86_64 architecture and
therefore use the according SWT binaries.

With this change, the used GitHub actions runner is changed from
macos-latest to macos-latest-large to ensure that still the x86_64 build
is used. This is supposed to restore the existing purpose of the
configuration. Changing to or adding an aarch64 runner for MacOS can be
done subsequently.

dependabot[bot] committed:
Bump eclipsecbi/fedora-gtk3-mutter in /container

Bumps eclipsecbi/fedora-gtk3-mutter from `317e341` to `3a588cf`.

---
updated-dependencies:
- dependency-name: eclipsecbi/fedora-gtk3-mutter
  dependency-type: direct:production
...

Signed-off-by: dependabot[bot] <support@github.com>

HeikoKlare committed:
Reduce Win32 FontRegistries visibility and move tests inside fragment

The classes DefaultSWTFontRegistry and ScalingSWTFontRegistry are
currently public only to make them testable from a test bundle. There is
currently no pre-configured way to implement unit tests for classes that
are not directly accessible via public API.

With this change, the SWT Win32 fragments are extended by a
configuration for providing fragment-internal unit tests executed with
plain Maven surefire, which can access the classes under test even with
reduced visibility. Due to current limitation in Tycho, an according
temporary extension to the Maven configuration for the binaries projects
is made. This configuration is applied to the tests for the
DefaultSWTFontRegistry and ScalingSWTFontRegistry, such that their
visibilities can properly be reduced to package visibility.

HeikoKlare committed:
Extend GitHub actions configuration matrix for MacOS AArch64 build

GitHub provides aarch64 runners for MacOS and has recently made them
even the default runners for MacOS. In order to also validate on aarch64
in addition to x86_64, this change extends the build configuration
matrix to also be executed on a MacOS aarch64 runner.

akoch-yatta commented on bundles/org.eclipse.swt/Eclipse SWT/win32/org/eclipse/swt/widgets/Composite.java:
```diff
@@ -1519,6 +1519,7 @@ LRESULT WM_PAINT (long wParam, long lParam) {
 				long hBufferedPaint = OS.BeginBufferedPaint (hDC, prcTarget, flags, null, phdc);
 				GCData data = new GCData ();
 				data.device = display;
+				data.deviceZoom = getZoom();
```
I think we should set data.nativeDeviceZoom as well here

akoch-yatta commented on bundles/org.eclipse.swt/Eclipse SWT/win32/org/eclipse/swt/widgets/Composite.java:
```diff
@@ -1547,6 +1548,7 @@ LRESULT WM_PAINT (long wParam, long lParam) {
 
 			/* Create the paint GC */
 			GCData data = new GCData ();
+			data.deviceZoom = getZoom();
```
I think we should set data.nativeDeviceZoom as well here

akoch-yatta commented on bundles/org.eclipse.swt/Eclipse SWT/win32/org/eclipse/swt/widgets/Composite.java:
```diff
@@ -1581,6 +1583,7 @@ LRESULT WM_PAINT (long wParam, long lParam) {
 					paintGC = gc;
 					gc = new GC (image, paintGC.getStyle() & SWT.RIGHT_TO_LEFT);
 					GCData gcData = gc.getGCData ();
+					gcData.deviceZoom = getZoom();
```
I think we should set data.nativeDeviceZoom as well here

akoch-yatta commented on bundles/org.eclipse.swt/Eclipse SWT/win32/org/eclipse/swt/widgets/Tree.java:
```diff
@@ -1050,6 +1051,7 @@ LRESULT CDDS_ITEMPREPAINT (NMTVCUSTOMDRAW nmcd, long wParam, long lParam) {
 			}
 			int nSavedDC = OS.SaveDC (hDC);
 			GCData data = new GCData ();
+			data.deviceZoom = getZoom();
```
I think we should set data.nativeDeviceZoom as well here

akoch-yatta commented on bundles/org.eclipse.swt/Eclipse SWT/win32/org/eclipse/swt/widgets/Widget.java:
```diff
@@ -52,7 +52,10 @@
 
-	private int zoom;
+	/**
+	 * @since 3.126
```
Extend the comment, e.g. have a look at the "handle" attribute

akoch-yatta commented on bundles/org.eclipse.swt/Eclipse SWT/win32/org/eclipse/swt/widgets/Shell.java:
```diff
@@ -125,7 +125,10 @@ public class Shell extends Decorations {
 	int minWidth = SWT.DEFAULT, minHeight = SWT.DEFAULT, maxWidth = SWT.DEFAULT, maxHeight = SWT.DEFAULT;
-	private int nativeZoom;
+	/**
+	 * @since 3.126
```
Extend the comment, e.g. have a look at the "handle" attribute

akoch-yatta commented on bundles/org.eclipse.swt/Eclipse SWT/win32/org/eclipse/swt/widgets/Tree.java:
```diff
@@ -831,6 +831,7 @@ LRESULT CDDS_ITEMPOSTPAINT (NMTVCUSTOMDRAW nmcd, long wParam, long lParam) {
 				int nSavedDC = OS.SaveDC (hDC);
 				GCData data = new GCData ();
 				data.device = display;
+				data.deviceZoom = getZoom();
```
I think we should set data.nativeDeviceZoom as well here

akoch-yatta commented on bundles/org.eclipse.swt/Eclipse SWT/win32/org/eclipse/swt/widgets/Label.java:
```diff
@@ -563,6 +563,7 @@ void wmDrawChildImage(DRAWITEMSTRUCT struct) {
 	}
 
 	GCData data = new GCData();
+	data.deviceZoom = getZoom();
```
I think we should set data.nativeDeviceZoom as well here

akoch-yatta commented on bundles/org.eclipse.swt/Eclipse SWT/win32/org/eclipse/swt/widgets/Control.java:
```diff
@@ -3657,6 +3657,19 @@ public void setRedraw (boolean redraw) {
 }
 
+/**
+ * @since 3.125
```
do we need a since here?

amartya4256 replied:
Nope

akoch-yatta commented on bundles/org.eclipse.swt/Eclipse SWT/win32/org/eclipse/swt/widgets/Control.java:
```diff
@@ -3657,6 +3657,19 @@ public void setRedraw (boolean redraw) {
+	if(event != null && event.gc != null && event.gc.getGCData() != null) {
+		event.gc.getGCData().deviceZoom = getZoom();
+		event.gc.getGCData().nativeDeviceZoom = getShell().getNativeZoom();
+		DPIUtil.setDeviceZoom(getShell().getNativeZoom());
```
For what did we need that line for?

akoch-yatta commented on tests/org.eclipse.swt.tests.win32/JUnit Tests/org/eclipse/swt/tests/win32/graphics/GCWin32Tests.java:
```diff
@@ -0,0 +1,65 @@
+package org.eclipse.swt.tests.win32.graphics;
```
copy the header comment here

akoch-yatta commented on bundles/org.eclipse.swt/Eclipse SWT/win32/org/eclipse/swt/graphics/Image.java:
```diff
@@ -138,8 +138,9 @@ public final class Image extends Resource implements Drawable {
+	 * @since 3.126
 	 */
-	private int currentDeviceZoom = 100;
+	public int currentDeviceZoom = 100;
```
do we need it still public?

amartya4256 replied:
Made it package protected since it is still needed in GC

akoch-yatta commented on bundles/org.eclipse.swt/Eclipse SWT/win32/org/eclipse/swt/graphics/GC.java:
```diff
@@ -4242,10 +4250,10 @@ void setClipping(long clipRgn) {
-	y = DPIUtil.autoScaleUp(drawable, y);
-	width = DPIUtil.autoScaleUp(drawable, width);
-	height = DPIUtil.autoScaleUp(drawable, height);
+	x = DPIUtil.autoScaleUp (drawable, x, getDeviceZoom());
```
remove the space in this and the following lines

akoch-yatta commented on bundles/org.eclipse.swt/Eclipse SWT/win32/org/eclipse/swt/graphics/GC.java:
```diff
@@ -3887,6 +3895,7 @@ long identity() {
 
 void init(Drawable drawable, GCData data, long hDC) {
 	int foreground = data.foreground;
+	this.data = data;
```
What was the reason we needed to move that assignment up?

amartya4256 replied:
cant remember haha

akoch-yatta commented on bundles/org.eclipse.swt/Eclipse SWT/win32/org/eclipse/swt/graphics/GC.java:
```diff
@@ -971,9 +979,9 @@ public void drawImage (Image image, int srcX, int srcY, int srcWidth, int srcHei
-	int deviceZoom = DPIUtil.getDeviceZoom();
+	Rectangle src = DPIUtil.autoScaleUp(drawable, new Rectangle(srcX, srcY, srcWidth, srcHeight), getDeviceZoom());
+	Rectangle dest = DPIUtil.autoScaleUp(drawable, new Rectangle(destX, destY, destWidth, destHeight), getDeviceZoom());
+	int deviceZoom = getDeviceZoom();
```
I would move that line up and use the deviceZoom Attribute instead of calling getDeviceZoom multiple times

akoch-yatta commented on bundles/org.eclipse.swt/Eclipse SWT/win32/org/eclipse/swt/graphics/GC.java:
```diff
@@ -455,8 +463,8 @@ void checkGC(int mask) {
 public void copyArea (Image image, int x, int y) {
-	x = DPIUtil.autoScaleUp(drawable, x);
-	y = DPIUtil.autoScaleUp(drawable, y);
+	x = DPIUtil.autoScaleUp(drawable, x, getDeviceZoom());
```
When you are using the zoom multiple times you should store it to an attribute before and use that. This way, we are sure we are definitely using the same value for zoom for all values
int deviceZoom = getDeviceZoom();
x = DPIUtil.autoScaleUp(drawable, x, deviceZoom);
y = DPIUtil.autoScaleUp(drawable, y, deviceZoom);

akoch-yatta commented on bundles/org.eclipse.swt/Eclipse SWT/win32/org/eclipse/swt/graphics/GC.java:
```diff
@@ -133,6 +134,13 @@ public final class GC extends Resource {
+	if(drawable instanceof Widget) {
+		data.deviceZoom = ((Widget) drawable).zoom;
+		if (drawable instanceof Control)
+			data.nativeDeviceZoom = ((Control) drawable).getShell().nativeZoom;
```
can there be an issue, when we don't set nativeDeviceZoom with Widget and Image? We should be sure about that

dependabot[bot] committed:
Bump eclipsecbi/fedora-gtk3-mutter in /container

Bumps eclipsecbi/fedora-gtk3-mutter from `3a588cf` to `fafd840`.

---
updated-dependencies:
- dependency-name: eclipsecbi/fedora-gtk3-mutter
  dependency-type: direct:production
...

Signed-off-by: dependabot[bot] <support@github.com>

HannesWell committed:
[Build] Simplify collecting native sources in Jenkins pipeline

Stash the native sources only once per OS, which reduces the runtime and
required storage-space on the Jenkins controller.

Commit:
v4965r9

fedejeanne committed:
Exclude Snippet123 from the classpath in Mac and Linux

The snippet is for Windows

dependabot[bot] committed:
Bump eclipsecbi/fedora-gtk3-mutter in /container

Bumps eclipsecbi/fedora-gtk3-mutter from `fafd840` to `fe48677`.

---
updated-dependencies:
- dependency-name: eclipsecbi/fedora-gtk3-mutter
  dependency-type: direct:production
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
Bump actions/checkout from 4.1.4 to 4.1.5

Bumps [actions/checkout](https://github.com/actions/checkout) from 4.1.4 to 4.1.5.
- [Release notes](https://github.com/actions/checkout/releases)
- [Changelog](https://github.com/actions/checkout/blob/main/CHANGELOG.md)
- [Commits](https://github.com/actions/checkout/compare/0ad4b8fadaa221de15dcec353f45205ec38ea70b...44c2b7a8a4ea60a981eaca3cf939b5f4305c123b)

---
updated-dependencies:
- dependency-name: actions/checkout
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

iloveeclipse committed:
Removing myself from win32 port owners

fredg02 committed:
Remove macos-latest-large label

See https://www.eclipse.org/lists/cross-project-issues-dev/msg19891.html for context

iloveeclipse committed:
FileDialog & DirectoryDialog hang after closing them with opened menu

`gtk_file_chooser_native_new handle()` should be dereferenced after use
as proposed in https://docs.gtk.org/gtk3/class.FileChooserNative.html.

Otherwise `gtk_native_dialog_run()` seem to continue to run (probably
because the opened menu still references the dialog), with the effect
that: `the recursive main loop gives the effect of a modal dialog (it
prevents the user from interacting with other windows in the same window
group while the dialog is run)`.

See https://docs.gtk.org/gtk3/method.NativeDialog.run.html

Fixes https://github.com/eclipse-platform/eclipse.platform.swt/issues/1062

HannesWell committed:
Revert "Make requirement of o.e.swt for swt.win32.aarch64 temporarily
optional"

Now that an Equinox launcher is  available for Windows on ARM and the
win32.aarch64 environment is activated in the Eclipse Maven build, the
requirement don't need to be optional anymore.

This reverts commit 5a79f9f370c4b378f46ef58b9e09e82f8729f7d9.

fedejeanne committed:
Update coactions to use NodeJS v20

GH Actions transitioned to Node 20 in Septmber 2023 and all projects
should transition too (see:
https://github.blog/changelog/2023-09-22-github-actions-transitioning-from-node-16-to-node-20/)

This gets rid of the warnings in the runs of GH Actions.

fedejeanne committed:
Update coactions to use Node 20

Use the commit indicated in
https://github.com/coactions/setup-xvfb/issues/29#issuecomment-2100951480

iloveeclipse committed:
Revert "Set the selection to 1st CTabItem when it is created."

This reverts commit 49c04a7f17f50cd667c20c6400099e1e5e06a717 because it
introduces regressions in client code.

See https://github.com/eclipse-platform/eclipse.platform.swt/issues/46#issuecomment-2107616850

amartya4256 committed:
Store nativeDeviceZoom in widgets instead of zoom for win32

This contribution is to provide the nativeDeviceZoom to all the widgets
so that it can be used later on for e.g. font scaling and more.
Currently, the nativeDeviceZoom is only available via the shell, which
is not sufficient later when there is no shell available, e.g. GC.

Contributes to #62 and #127

Phillipus committed:
Fix for Mac drawing bug issue 1012 part 2 (#1232)

- OS.objc_msgSend(id, OS.sel_setClipsToBounds_, true) needs to be applied in two more cases

- See https://github.com/eclipse-platform/eclipse.platform.swt/issues/1012

hmartinez82 committed:
[win32][arm64] Support Dark Theme on newer Windows builds

dependabot[bot] committed:
Bump actions/checkout from 4.1.5 to 4.1.6

Bumps [actions/checkout](https://github.com/actions/checkout) from 4.1.5 to 4.1.6.
- [Release notes](https://github.com/actions/checkout/releases)
- [Changelog](https://github.com/actions/checkout/blob/main/CHANGELOG.md)
- [Commits](https://github.com/actions/checkout/compare/44c2b7a8a4ea60a981eaca3cf939b5f4305c123b...a5ac7e51b41094c92402da3b24376905380afc29)

---
updated-dependencies:
- dependency-name: actions/checkout
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

Phillipus committed:
Fix warning message on Mac about secure coding not enabled (#1231)

- See https://github.com/eclipse-platform/eclipse.platform.swt/issues/1228

- On macOS 14 and later a warning message is written to console:

"WARNING: Secure coding is not enabled for restorable state! Enable secure coding by implementing NSApplicationDelegate.applicationSupportsSecureRestorableState: and returning YES."

- As recommended by Apple, this change adds a new selector for applicationSupportsSecureRestorableState and returns a 1 value (YES) in the callback in the Display class

- However, this only takes care of the NSApplicationDelegate created in the Display class. The warning message will still appear when a splash screen is created in Equinox JNI code so a fix there is also required. See https://github.com/eclipse-equinox/equinox/issues/630

- This implementation for SWT can be tested by using the -noSplash argument when launching Eclipse

Phillipus committed:
Revert "Fix warning message on Mac about secure coding not enabled" (#1240)

- This can cause a crash to desktop

iloveeclipse committed:
Update bug_report.md for Linux users

Added note for Linux users about crash reporting/webkit

See https://github.com/eclipse-platform/eclipse.platform.swt/issues/843

Commit:
v4965r10

deepika-u committed:
SWT Javadoc bash changes for 4.32

Fixes https://github.com/eclipse-platform/eclipse.platform.swt/issues/1245

akoch-yatta committed:
Reset tree item height on DPI change

With the current implementation of the handler on DPI zoom changes, not all rescaling cases are covered correctly. If the item height was set at least once programmatically (currently done in Tree::setItemHeight), it will never be done by the OS anymore. Resetting itto -1 will re-enable the default behavior again.

Contributes to #62 and #127

HannesWell committed:
Update JDT compiler settings in example and test projects

Ensure for all example and test project that all current preference keys
are contained in the preference files.

HannesWell committed:
Unify JDT compiler settings of example and test projects

In case of conflicts the stricter variant is chosen, i.e. 'error' over
'warning' and 'warning' over 'info' or not to ignore a problem instead
of ignoring it.
Exceptions are
1. Problems about unused elements because tests and examples
have some unused elements for good reasons and only warnings be
suppressed. Since the build fails on new warnings it is hard to
introduce new ones.
2. Problems about indirect static access are set to info because some
examples are better to read if static members are accessed through
sub-classes.

Fix or suppress all new warnings.

HannesWell committed:
Link to shared JDT compiler settings file in example and test projects

Commit:
v4965r11

dependabot[bot] committed:
Bump eclipsecbi/fedora-gtk3-mutter in /container

Bumps eclipsecbi/fedora-gtk3-mutter from `fe48677` to `02371de`.

---
updated-dependencies:
- dependency-name: eclipsecbi/fedora-gtk3-mutter
  dependency-type: direct:production
...

Signed-off-by: dependabot[bot] <support@github.com>

akoch-yatta committed:
Remove methods producing pixel values in GlyphMetrics

In this contribution the DPIUtil dependency from org.eclipse.swt.graphics.GlyphMetrics is removed including all package protected pixel producing methods. As (at least) Windows receivces HiDPI support for multiple zoom levels, this is a prerequisite to contribute this feature. As a consequence all callers inside SWT that still used the pixel dependent methods are adapt to only use the public API.

Contributes to #62 and #131

MohananRahul committed:
POM and product version changes for 4.33 release (#1257)

amartya4256 committed:
added OS level API bindings for win32

This commit provides a native API binding for win32 to get the ancestor of the
provided window.

Contributes to #62 and #127

deepika-u committed:
Configure SWT build scripts for 4.33

Fixes https://github.com/eclipse-platform/eclipse.platform.releng.aggregator/issues/2071

Updated REVISION number as per this review comment to zero.
https://github.com/eclipse-platform/eclipse.platform.swt/pull/1078/files#r1521096108

@niraj-modi : Can you review and merge this pr please.

iloveeclipse committed:
Bump SWT bundle versions for 4.33

See https://github.com/eclipse-platform/eclipse.platform.releng.aggregator/issues/2085

Commit:
v4966r1

HannesWell committed:
Version bump for 4.33 stream

HeikoKlare committed:
Remove API filter for 4.32 release

For the 4.32 release, an API filter for the TypedListener being
annotated with `@noextend` has been introduced. After having switched to
the 4.33 stream, this filter now became obsolete and is thus removed
with this change.

See https://github.com/eclipse-platform/eclipse.platform.swt/pull/1101

Commit:
v4966r2
